src/lib/operators/abstract_operator.cpp

```diff
@@ -74,15 +74,15 @@ std::shared_ptr<const Table> AbstractOperator::_input_table_left() const { retur
 
 std::shared_ptr<const Table> AbstractOperator::_input_table_right() const { return _input_right->get_output(); }
 
-std::shared_ptr<TransactionContext> AbstractOperator::transaction_context() const {
+bool AbstractOperator::transaction_context_is_set() const {
   // https://stackoverflow.com/questions/45507041/how-to-check-if-weak-ptr-is-empty-non-assigned
-  DebugAssert(
-      [context = _transaction_context]() {
-        bool transaction_context_set = context.owner_before(std::weak_ptr<TransactionContext>{}) ||
-                                       std::weak_ptr<TransactionContext>{}.owner_before(context);
-        return !transaction_context_set || !context.expired();
-      }(),
-      "TransactionContext is expired, but SQL Query Executor should still own it (Operator: " + name() + ")");
+  return _transaction_context.owner_before(std::weak_ptr<TransactionContext>{}) ||
```

Comment:
I'd much prefer just wrapping the weak_ptr in an optional to check whether it is set. Or does this do something more than that?

Reply:
This would add lots of `if(context)` checks everywhere else, because we use the feature that weak_ptr.lock returns a nullptr if empty. But if we wrap it in an optional, we need to dereference it each time which will fail if it is not set.

Reply:
As a third option, we could either add a `bool` flag `context_is_set` or we use `optional::value_or_default(nullptr)` in `AbstractOperator::transaction_context()`